balancer/endpointsharding/endpointsharding_test.go

```diff
@@ -157,3 +159,91 @@ func (s) TestEndpointShardingBasic(t *testing.T) {
 		t.Fatalf("error in expected round robin: %v", err)
 	}
 }
+
+// TestInitialChannelStateWaitingForFirstResolverUpdate verifies the initial state of the channel
+// when a manual name resolver doesn't provide any updates.
+func (s) TestInitialChannelStateWaitingForFirstResolverUpdate(t *testing.T) {
+	backend := stubserver.StartTestService(t, nil)
+	defer backend.Stop()
+
+	mr := manual.NewBuilderWithScheme("e2e-test")
+	defer mr.Close()
+
+	mr.InitialState(resolver.State{
+		Endpoints: []resolver.Endpoint{},
+	})
+
+	cc, err := grpc.NewClient(mr.Scheme()+":///", grpc.WithResolvers(mr), grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		t.Fatalf("Failed to create new client: %v", err)
+	}
+	defer cc.Close()
+
+	if state := cc.GetState(); state != connectivity.Idle {
+		t.Fatalf("Expected initial state Idle, got %v", state)
+	}
+
+	cc.Connect()
+	state := cc.GetState()
+	if state != connectivity.Connecting && state != connectivity.TransientFailure {
+		t.Fatalf("Expected state Connecting or TransientFailure after Connect(), got %v", state)
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	// Allow the state to transition either to Connecting or TransientFailure.
+	if state == connectivity.Connecting {
+		testutils.AwaitState(ctx, t, cc, connectivity.TransientFailure)
+	}
+
+	errChan := make(chan error, 1)
```

Comment:
No need to use a channel to propagate the error, you can call `t.Errorf()` and log the error in the go routine itself.